polus-imagenet-model-featurization-plugin/src/main.py

```diff
@@ -0,0 +1,142 @@
+import re, sys, os
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import tensorflow as tf
+import argparse, logging
+from skimage import io, transform
+
+valid_models =  [
+    'Xception',
+    'VGG16',
+    'VGG19',
+    'ResNet50',
+    'ResNet101',
+    'ResNet152',
+    'ResNet50V2',
+    'ResNet101V2',
+    'ResNet152V2',
+    'InceptionV3',
+    'InceptionResNetV2',
+    'DenseNet121',
+    'DenseNet169',
+    'DenseNet201'
+]
+
+
+def get_imagenet_model(model):
+    model_method = getattr(tf.keras.applications, model)
+    return model_method(weights='imagenet', include_top=False, pooling='avg')
+
+
+def load_img(image_path, target_size=None):
+    img = io.imread(image_path)
+    dim = len(img.shape)
+    
+    # Only grayscale or RGB
+    if dim not in [2, 3]:
+        return None 
+    
+    # Only three channels. 
+    if (dim == 3) and (img.shape[-1] != 3):
+        return None
+    
+    # If the image is grayscale, expand to three dims.
+    if dim == 2:
+        img = np.stack((img,)*3, axis=-1)
+
+    if target_size is not None:
+        img = transform.resize(img, output_shape=target_size, anti_aliasing=True)
+    
+    return img
+
+
+if __name__=='__main__':
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger('main')
+    logger.setLevel(logging.INFO)
+
+    ''' Argument parsing '''
+    logger.info('Parsing arguments...')
+    parser = argparse.ArgumentParser(prog='main', description='Image featurization using models pre-trained on ImageNet')
+    
+    # Input arguments
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input image collection to be processed by this plugin', required=True)
+    parser.add_argument('--model', dest='model', type=str,
+                        help='Pre-trained ImageNet model to use for featurization', required=True)
+    parser.add_argument('--resolution', dest='resolution', type=str,
+                        help='Resolution to which the input images are scaled', required=False)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+
+    input_dir = args.inpDir
+    logger.info('inpDir = {}'.format(input_dir))
```

Comment:
There is a quirk in WIPP where we need to check to see if there is an `images` subdirectory, and switch to it if it exists.

Reply:
Will fix, thanks.